packages/client/src/raffles/types/postRaffleParticipation.types.ts

```diff
@@ -0,0 +1,23 @@
+import type { Config } from '../..';
+import type {
+  Raffle,
+  RaffleParticipation,
+  RaffleParticipationStatus,
+} from './raffles.types';
+
+export type PostParticipationData = {
+  userId: string;
+  billingAddressId: string;
+  shippingAddressId: string;
+  productVariantId: string;
+  paymentTokenId: string | null;
+  trackingCorrelationId: string | null;
+};
+
+export type PostRaffleParticipation = (
+  raffleId: Raffle['id'],
+  data: PostParticipationData,
+  config?: Config,
+) => Promise<
+  RaffleParticipation & { status: RaffleParticipationStatus.PARTICIPATED }
```

Comment:
Is this `status` making sense? I mean, besides the swagger returning the same enum, why would it always return "Participated"? If you POST a `paymentTokenId` to a "Winner" participation, would it return "Participated"? 🤔 

Do you understand what I'm trying to say?

On a side topic, to override a property you need to omit it first - something like this:

```js
Omit<RaffleParticipation, 'status'> & { status: RaffleParticipationStatus.PARTICIPATED }
```

Reply:
done. pls double check